ftl/python/builder.py

```diff
@@ -43,60 +43,77 @@ def __init__(self, ctx, args, cache_version_str):
     def Build(self):
         lyr_imgs = []
         lyr_imgs.append(self._base_image)
+
         if ftl_util.has_pkg_descriptor(self._descriptor_files, self._ctx):
+            # check cache or build interpreter layer
             interpreter_builder = package_builder.InterpreterLayerBuilder(
                 self._venv_dir, self._python_cmd, self._venv_cmd)
             cached_int_img = None
             if self._args.cache:
-                with ftl_util.Timing("checking cached int layer"):
+                with ftl_util.Timing("checking cached interpreter layer"):
                     key = interpreter_builder.GetCacheKey()
                     cached_int_img = self._cache.Get(key)
             if cached_int_img is not None:
                 interpreter_builder.SetImage(cached_int_img)
             else:
-                with ftl_util.Timing("building int layer"):
+                with ftl_util.Timing("building interpreter layer"):
                     interpreter_builder.BuildLayer()
                 if self._args.cache:
-                    with ftl_util.Timing("uploading int layer"):
+                    with ftl_util.Timing("uploading interpreter layer"):
                         self._cache.Set(interpreter_builder.GetCacheKey(),
                                         interpreter_builder.GetImage())
             lyr_imgs.append(interpreter_builder)
 
-            pkg_descriptor = ftl_util.descriptor_parser(
-                self._descriptor_files, self._ctx)
+            # check cache or build package layers
+            req_txt_builder = package_builder.PackageLayerBuilder(
+                self._ctx, self._descriptor_files, None,
+                interpreter_builder)
+            cached_req_txt_img = None
+            if self._args.cache:
```

Comment:
just a note for the future - it would be nice to move this _args.cache check into the cache object itself so we don't have to remember to check this everywhere.

Reply:
added a pr